plugins/in_node_exporter_metrics/ne_thermalzone_linux.h

```diff
@@ -0,0 +1,34 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2015-2022 The Fluent Bit Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef FLB_IN_NE_THERMALZONE_LINUX_H
+#define FLB_IN_NE_THERMALZONE_LINUX_H
+
+#include "ne.h"
+
+#define THERMAL_ZONE_BASE "/class/thermal/thermal_zone"
+#define THERMAL_ZONE_PATTERN THERMAL_ZONE_BASE "[0-9]*"
+#define COOLING_DEVICE_BASE "/class/thermal/cooling_device"
+#define COOLING_DEVICE_PATTERN COOLING_DEVICE_BASE "[0-9]*"
+
+int ne_thermalzone_init(struct flb_ne *ctx);
+int ne_thermalzone_update_thermal_zones(struct flb_ne *ctx);
+int ne_thermalzone_update_cooling_devices(struct flb_ne *ctx);
```

Comment:
Shouldn't we take care of macOS(darwin)? We already has merged https://github.com/fluent/fluent-bit/pull/7947 PR which supports macOS. So, we have to disable this metrics on macOS.

Reply:
The same needs to be done for at least:
- ne_stat_linux.c
- ne_vmstat_linux.c
- ne_filefd_linux.c

Reply:
I copied ne_filesystem.[c,h] to make the thermal_zone implementation cross platform capable and leave room for other implementations on other OSes.